src/core/sorted_map.cc

```diff
@@ -573,34 +573,50 @@ size_t SortedMap::DeleteRangeByLex(const zlexrangespec& range) {
 }
 
 SortedMap::ScoredArray SortedMap::PopTopScores(unsigned count, bool reverse) {
+  DCHECK_GT(count, 0u);
   DCHECK_EQ(score_map->UpperBoundSize(), score_tree->Size());
   size_t sz = score_map->UpperBoundSize();
 
   ScoredArray res;
 
-  if (sz == 0)
+  DCHECK_GT(sz, 0u);  // Empty sets are not allowed.
+
+  if (sz == 0 || count == 0)
     return res;
 
-  if (count >= sz)
-    count = score_map->UpperBoundSize();
+  if (count > sz)
+    count = sz;
 
   res.reserve(count);
+
+  auto cb = [&](ScoreSds obj) {
+    res.emplace_back(string{(sds)obj, sdslen((sds)obj)}, GetObjScore(obj));
+
+    // We can not delete from score_tree because we are in the middle of the iteration.
+    CHECK(score_map->Erase((sds)obj));
+    return true;
```

Comment:
so `return false` was just saved for a future pr?

Reply:
what do you mean?